qa/sc_evm_bootstrap.py

```diff
@@ -0,0 +1,253 @@
+#!/usr/bin/env python3
+import json
+import logging
+from decimal import Decimal
+
+from eth_utils import to_checksum_address
+
+from SidechainTestFramework.sc_boostrap_info import SCNodeConfiguration, SCCreationInfo, MCConnectionInfo, \
+    SCNetworkConfiguration, LARGE_WITHDRAWAL_EPOCH_LENGTH
+from SidechainTestFramework.sc_test_framework import SidechainTestFramework
+from httpCalls.wallet.allPublicKeys import http_wallet_allPublicKeys
+from httpCalls.wallet.createPrivateKeySecp256k1 import http_wallet_createPrivateKeySec256k1
+from test_framework.util import assert_equal, assert_true, start_nodes, \
+    websocket_port_by_mc_node_index, forward_transfer_to_sidechain
+from SidechainTestFramework.scutil import bootstrap_sidechain_nodes, \
+    start_sc_nodes, is_mainchain_block_included_in_sc_block, \
+    check_mainchain_block_reference_info, \
+    AccountModelBlockVersion, EVM_APP_BINARY, generate_next_blocks, generate_next_block, generate_account_proposition, \
+    convertZenniesToWei, convertZenToZennies, connect_sc_nodes, computeForgedTxFee
+from SidechainTestFramework.account.httpCalls.wallet.balance import http_wallet_balance
+
+"""
+Check the EVM bootstrap feature.
+
+Configuration: bootstrap 1 SC node and start it with genesis info extracted from a mainchain node.
+    - Mine some blocks to reach hard fork
+    - Create 1 SC node
+    - Extract genesis info
+    - Start SC node with that genesis info
+
+Test:
+    For the SC node:
+        - verify the MC block is included
+"""
+class SCEvmBootstrap(SidechainTestFramework):
+
+    sc_nodes_bootstrap_info=None
+    number_of_mc_nodes = 1
+    number_of_sidechain_nodes = 2
+    API_KEY = "Horizen"
+
+    def setup_nodes(self):
+        return start_nodes(self.number_of_mc_nodes, self.options.tmpdir)
+
+    def sc_setup_network(self, split = False):
```

Comment:
Some parts of the initialization are identical in almost all of the Python tests. We could refactor this to a default setup for all tests that don't require a custom setup.